### DISCUSSION TITLE
Why is tanstack router not updating routes?

### DISCUSSION THREAD

**Discussion by Happily-Coding** [fastapi/full-stack-fastapi-template#1180] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
No code is specially relevant.
```

### Description

I'm trying to add new routes to the project, but I found that tanstack routes aren't being updated.

routeTree.gen.ts needs should be automatically updated for any modification to the routes to work, but it isn't.

How can we verify the problem is that routeTree.gen.ts is not being updated?
1.Setup a brand new copy of the project
2.Run docker compose up --build . The project will build.
3.Delete routeTree.gen.ts . 
4.Run docker compose up --build . The project should re-generate the original file if everything was working properly. It doesn't. The project won't build.

I realize i could npm install @tanstack/router-cli --save-dev and call npx tsr generate to re-generate the file, but in my opinion what would be correct is for the build process to generate the correct file, and in fact I believe it should with the current setup. This [issue ](https://github.com/TanStack/router/discussions/1218) could help in achieving that, but just adding that to package.json doesn't work, ofc, We probably need to modify other parts of the package.json or of the dockerfiles, but as of right now i'm unsure which/how in order to achieve the best architechture.

### Operating System

Windows

### Operating System Details

I don't think more details are relevant.

### Python Version

3.12.1

**Comment by Happily-Coding** [comment, marked as answer]
For anyone that comes across this post:

The reason that tan stack router is not updating routes is that docker compose up --build doesnt run the frontend in development mode, if running the project without docker, following the steps located [here](https://github.com/tiangolo/full-stack-fastapi-template/tree/master/frontend) , the routes will be properly updated.

It makes sense that the plugin doesn't update routes when not in development mode, since otherwise the build you deploy could be different from the build you tested in development, so I don't suggest fixing the issue since its behavior is correct.

That said, I assumed that the docker compose file was meant to be used for local development, unless explicitly overriden, and that doesn't seem to be the case, it seems the author intends development with this template to be done containerless, and so, the docker-compose is actually a local deployment composition.

TLDR: For development run the individual parts.
